common4j/src/main/com/microsoft/identity/common/java/crypto/IDecryptor.java

```diff
@@ -49,4 +49,20 @@ byte[] decrypt(@NonNull final Key key,
                    @NonNull final String decryptAlgorithm,
                    final byte[] iv,
                    final byte[] dataToBeDecrypted) throws ClientException;
+
+    /**
+     * Decrypt the given byte array.
+     *
+     * @param key               the key to decrypt with.
+     * @param decryptAlgorithm  algorithm to decrypt with.
+     * @param iv                an initialization vector (IV).
+     * @param dataToBeDecrypted the data to be encrypted.
+     * @return a decrypted byte array.
+     */
+    byte[] decrypt(@NonNull final Key key,
+                   @NonNull final String decryptAlgorithm,
+                   final byte[] iv,
+                   final byte[] dataToBeDecrypted,
```

Comment:
>final byte[] dataToBeDecrypted, [](http://example.com/codeflow?start=19&length=31)

why there's no tag parameter here?
Is it expected from caller to send dataToBeEncrypted with tag inside at?

Reply:
I mean I guess the answer to your question would be that `dataToBeEncrypted` is meant to be exact encrypted data produced through an AES GCM encryption. While decryption we don't need to know the tag, we just need to know the length of the tag. I would think that eSTS will supply us the tag length. Let me know if I missing something.

Reply:
yes, we need tag as well to perform decryption. 
Response from ests has format
[Header].[EncryptedKey].[IV].[Payload].[Tag]

At the end data that needs to decrypted is payload+tag, so we need to concatenate them and tag length is used in setting up the spec.
If we pass payload and tag here, we would concatenate them and decrypt it. 
With the signature you have now, such concatenation would be exacted by caller.

Reply:
Okay, so if I understand correctly, the payload that eSTS sends us (that needs to be decrypted) doesn't actually include the tag? They seem to be sending us the raw tag separately per your response example. eSTS wants us to include the tag and then do the decryption?

I mean I can modify the interface, but the current interface is designed to be more generic encrypt / decrypt interface and NOT specific to the mentioned eSTS scenario. From what I saw in examples online, it seemed the cipherText given to the AES GCM decrypt logic would include the tag and thus while decryption the only thing needed to be known was the tag length and so I followed similar logic in my implementation.

Reply:
For now, I'm keeping this interface generic. We can build a wrapper around this for purposes of your specific scenario to perform the concatenation.

Reply:
Yes, they send tag separately. 
I get your point. I see that with Java Cipher/GCMParameterSpec the tag is generated internally and is part of final ciphertext you get after encryption. 
So, it seems fine to keep decrypt analogous. We combine text and tag in a layer above.